bigquery_etl/cli/stage.py

```diff
@@ -226,8 +229,11 @@ def _view_dependencies(artifact_files, sql_dir):
             for dependency in view.table_references:
                 dependency_components = dependency.split(".")
                 if dependency_components[2:3] == ["INFORMATION_SCHEMA"]:
-                    # INFORMATION_SCHEMA has more components that can be ignored here
-                    dependency_components = dependency_components[:3]
+                    # INFORMATION_SCHEMA has more components that will be treated as the table name
+                    # no deploys for INFORMATION_SCHEMA will happen later on
+                    dependency_components = dependency_components[:2] + [
+                        ".".join(dependency_components[2:])
+                    ]
```

Comment:
This will ensure that any references to `INFORMATION_SCHEMA` get set to the `INFORMATION_SCHEMA` in the stage project

Reply:
This handles `{project}.{region}.INFORMATION_SCHEMA` and `{project}.{dataset}.INFORMATION_SCHEMA` references, but for any `{region}.INFORMATION_SCHEMA` or `{dataset}.INFORMATION_SCHEMA` references then the region/dataset will end up being considered the project and `INFORMATION_SCHEMA` will be considered the dataset.

A possible alternative approach here would be to detect such cases and fill in the project based on the view (before this `if` statement):
```py
                if dependency_components[1:2] == ["INFORMATION_SCHEMA"]:
                    dependency_components.insert(0, view.project)
```
Then `INFORMATION_SCHEMA` should always end up as part of the name, not the dataset, and my other suggested change wouldn't be necessary.  However, I'm not sure if that'd mess with how dependencies are matched.